Index: config/locales/en.yml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- config/locales/en.yml	(revision 4c16aa3aab9ad3110fd0e86dc9188fb287b18b59)
+++ config/locales/en.yml	(revision )
@@ -15,6 +15,7 @@
   debit_note: Nota de débito
   free: Libre
   circles: Círculos
+  mini: Mini-Ciclo
   activerecord:
     models:
       account:
Index: app/views/carts/show.html.erb
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/views/carts/show.html.erb	(revision 4c16aa3aab9ad3110fd0e86dc9188fb287b18b59)
+++ app/views/carts/show.html.erb	(revision )
@@ -15,10 +15,9 @@
       </div>
     <% end %>
   </div>
-  <%= @ciclo_actual_include %>
 	<% if @ciclo_actual.present? %>
-			<% if current_usuario && current_usuario.circulo && @ciclo_actual.tipo != 'free' %>
-					<% if @ciclo_actual.circulos.include?(current_usuario.circulo)  %>
+			<% if current_usuario && @ciclo_actual.tipo != 'free' || @ciclo_actual.tipo == 'mini' %>
+					<% if @ciclo_actual.circulos.include?(current_usuario.circulo) || @ciclo_actual.tipo == 'mini'  %>
 							<% if !current_usuario.pedido?(@ciclo_actual) %>
 									<%= render 'carts/cart' %>
 							<% else %>
Index: app/assets/javascripts/compras.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/assets/javascripts/compras.js	(revision 4c16aa3aab9ad3110fd0e86dc9188fb287b18b59)
+++ app/assets/javascripts/compras.js	(revision )
@@ -105,7 +105,7 @@
 	});
 
 	function showHideCirculos(valor){
-		if (valor == 'free'){
+		if (valor == 'free' || valor == 'mini'){
 			$('#content-circulos').addClass('hide');
 		}else{
 			$('#content-circulos').removeClass('hide');
Index: app/views/carts/_cart.html.erb
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/views/carts/_cart.html.erb	(revision 4c16aa3aab9ad3110fd0e86dc9188fb287b18b59)
+++ app/views/carts/_cart.html.erb	(revision )
@@ -79,7 +79,7 @@
 				</tbody>
 			</table>
 		</div>
-		<% if @ciclo_actual && current_usuario && current_usuario.circulo && !current_usuario.pedido?(Compra:@ciclo_actual) %>
+		<% if @ciclo_actual && current_usuario && !current_usuario.pedido?(Compra:@ciclo_actual) %>
 				<div class="row text-right">
 					<%= link_to('Finalizar Compra', create_pedido_path, method: :get, class: "btn btn-success #{'disabled' if @missing.present?}",
 											id: 'finalizar') %>
Index: app/controllers/carts_controller.rb
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/controllers/carts_controller.rb	(revision 4c16aa3aab9ad3110fd0e86dc9188fb287b18b59)
+++ app/controllers/carts_controller.rb	(revision )
@@ -81,7 +81,7 @@
 																	 warehouses_id: warehouse
 						).save
 					end
-					if ciclo.tipo != 'free'
+					if ciclo.tipo == 'circles'
 						if delivery.blank?
 							Sector.all.each do |sector|
 								if sector.id == Sector::CONSUMERS
Index: app/models/compra.rb
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/models/compra.rb	(revision 4c16aa3aab9ad3110fd0e86dc9188fb287b18b59)
+++ app/models/compra.rb	(revision )
@@ -4,7 +4,7 @@
   has_many :circulos, :through => :deliveries
   has_many :pedidos
 
-	enum tipo: [:circles, :free]
+	enum tipo: [:circles, :free, :mini]
 
 	validates :nombre, :descripcion, :fecha_inicio_compras, :fecha_fin_compras,
             :fecha_fin_pagos, :fecha_entrega_compras, presence: true
Index: app/views/compras/_form_circulos.html.erb
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/views/compras/_form_circulos.html.erb	(revision 4c16aa3aab9ad3110fd0e86dc9188fb287b18b59)
+++ app/views/compras/_form_circulos.html.erb	(revision )
@@ -1,4 +1,4 @@
-<div id="content-circulos" class="<%= 'hide' if @compra.tipo == 'free' %> ">
+<div id="content-circulos" class="<%= 'hide' if @compra.tipo == 'free' or @compra.tipo == 'mini' %> ">
 	<div class="panel panel-default">
 		<div class="panel-heading">
 			<h3 class="panel-title">Círculos</h3>
Index: app/views/compras/_form.html.erb
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/views/compras/_form.html.erb	(revision 4c16aa3aab9ad3110fd0e86dc9188fb287b18b59)
+++ app/views/compras/_form.html.erb	(revision )
@@ -33,6 +33,12 @@
 				Libre
 			</label>
 		</div>
+    <div class="radio">
+      <label>
+        <%= f.radio_button :tipo, 'mini' , checked: (@compra.tipo == 'mini') ? true : false  %>
+        Mini Ciclo
+      </label>
+    </div>
 	</div>
   <div class="form-group">
     <%= f.label :descripcion %><br>
Index: app/views/productos/_products.html.erb
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/views/productos/_products.html.erb	(revision 4c16aa3aab9ad3110fd0e86dc9188fb287b18b59)
+++ app/views/productos/_products.html.erb	(revision )
@@ -45,7 +45,7 @@
 						<p>Precio hiper <%= number_to_currency(precio_super) %></p>
 						<p>Ahorro <%= number_to_percentage(producto.ahorro) %></p>
 						<% if @ciclo_actual and usuario_signed_in? %>
-								<% if @ciclo_actual.circulos.include?(current_usuario.circulo) %>
+								<% if @ciclo_actual.circulos.include?(current_usuario.circulo) or @ciclo_actual.tipo == 'mini'%>
 										<%
                       if producto.stock.present? and producto.cantidad_permitida > producto.stock
                         cantidad = producto.stock
