Index: app/views/compras/_form.html.erb
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/views/compras/_form.html.erb	(revision 0cfc9c1470f26d75d957f378501fd19fcafb4eda)
+++ app/views/compras/_form.html.erb	(revision )
@@ -23,7 +23,7 @@
 		<%= f.label :tipo %>
 		<div class="radio">
 			<label>
-				<%= f.radio_button :tipo, 'circulos', checked: (@compra.tipo == 0 || @compra.tipo.nil?) ? true : false%>
+				<%= f.radio_button :tipo, 'circles', checked: (@compra.tipo == 'circles' || @compra.tipo.nil?) ? true : false%>
 				Círculos
 			</label>
 		</div>
Index: app/controllers/compras_controller.rb
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/controllers/compras_controller.rb	(revision 0cfc9c1470f26d75d957f378501fd19fcafb4eda)
+++ app/controllers/compras_controller.rb	(revision )
@@ -139,29 +139,29 @@
         descripcion: filtered_params[:descripcion],
 				tipo: filtered_params[:tipo],
         circulo_ids: filtered_params["circulo_ids"],
-        fecha_inicio_compras: Time.utc(
+        fecha_inicio_compras: Time.zone.local(
           filtered_params[:fecha_inicio_compras][:year],
           filtered_params[:fecha_inicio_compras][:month],
           filtered_params[:fecha_inicio_compras][:day],
           filtered_params[:fecha_inicio_compras][:hour],
-          filtered_params[:fecha_inicio_compras][:min]),
-        fecha_fin_compras: Time.utc(
+          filtered_params[:fecha_inicio_compras][:minute]),
+        fecha_fin_compras: Time.zone.local(
           filtered_params[:fecha_fin_compras][:year],
           filtered_params[:fecha_fin_compras][:month],
           filtered_params[:fecha_fin_compras][:day],
           filtered_params[:fecha_fin_compras][:hour],
-          filtered_params[:fecha_fin_compras][:min]),
-        fecha_fin_pagos: Time.utc(
+          filtered_params[:fecha_fin_compras][:minute]),
+        fecha_fin_pagos: Time.zone.local(
           filtered_params[:fecha_fin_pagos][:year],
           filtered_params[:fecha_fin_pagos][:month],
           filtered_params[:fecha_fin_pagos][:day],
           filtered_params[:fecha_fin_pagos][:hour],
-          filtered_params[:fecha_fin_pagos][:min]),
-        fecha_entrega_compras: Time.utc(
+          filtered_params[:fecha_fin_pagos][:minute]),
+        fecha_entrega_compras: Time.zone.local(
           filtered_params[:fecha_entrega_compras][:year],
           filtered_params[:fecha_entrega_compras][:month],
           filtered_params[:fecha_entrega_compras][:day],
           filtered_params[:fecha_entrega_compras][:hour],
-          filtered_params[:fecha_entrega_compras][:min])}
+          filtered_params[:fecha_entrega_compras][:minute])}
     end
 end
Index: app/views/layouts/_main_menu.html.erb
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/views/layouts/_main_menu.html.erb	(revision 0cfc9c1470f26d75d957f378501fd19fcafb4eda)
+++ app/views/layouts/_main_menu.html.erb	(revision )
@@ -52,11 +52,21 @@
             </li>
             <li class="col-sm-3">
               <ul>
-                <li class="dropdown-header">Features</li>
-                <li><a href="#">Auto Carousel</a></li>
-                <li><a href="#">Carousel Control</a></li>
-                <li><a href="#">Left & Right Navigation</a></li>
-                <li><a href="#">Four Columns Grid</a></li>
+                <li class="dropdown-header">Personas y Círculos</li>
+                <% if can? [:manage], circulos_path %>
+                    <li>
+                      <%= link_to(circulos_path) do %>
+                        Círculos
+                      <% end %>
+                    </li>
+                <% end %>
+                <% if can? [:manage], usuarios_path %>
+                    <li>
+                      <%= link_to(usuarios_path) do %>
+                          <span class="icon-key2"></span>Usuarios
+                      <% end %>
+                    </li>
+                <% end %>
                 <li class="divider"></li>
                 <li class="dropdown-header">Fonts</li>
                 <li><a href="#">Glyphicon</a></li>
@@ -65,20 +75,40 @@
             </li>
             <li class="col-sm-3">
               <ul>
-                <li class="dropdown-header">Plus</li>
-                <li><a href="#">Navbar Inverse</a></li>
-                <li><a href="#">Pull Right Elements</a></li>
-                <li><a href="#">Coloured Headers</a></li>
-                <li><a href="#">Primary Buttons & Default</a></li>
+                <li class="dropdown-header">Reportes</li>
+                <% if can? [:manage], geo_reports_path %>
+                    <li>
+                      <%= link_to(geo_reports_path) do %>
+                        <span class="fa fa-area-chart"></span>Mapa Reporte
+                      <% end %>
+                    </li>
+                <% end %>
+                <% if can? [:manage], reports_path %>
+                    <li>
+                      <%= link_to(reports_path) do %>
+                        <span class="fa fa-area-chart"></span>Productos por Proveedores
+                      <% end %>
+                    </li>
+                <% end %>
               </ul>
             </li>
             <li class="col-sm-3">
               <ul>
-                <li class="dropdown-header">Much more</li>
-                <li><a href="#">Easy to Customize</a></li>
-                <li><a href="#">Calls to action</a></li>
-                <li><a href="#">Custom Fonts</a></li>
-                <li><a href="#">Slide down on Hover</a></li>
+                <li class="dropdown-header">Administración</li>
+                  <% if can? [:manage], pedidos_path %>
+                    <li>
+                      <%= link_to(pedidos_path) do %>
+                        Pedidos
+                      <% end %>
+                    </li>
+                  <% end %>
+                <% if can? [:manage], accounts_path %>
+                  <li>
+                    <%= link_to(accounts_path) do %>
+                      <span class="icon-stats"></span>Cuentas Corrientes
+                    <% end %>
+                  </li>
+                <% end %>
               </ul>
             </li>
           </ul>
\ No newline at end of file
