Index: Gemfile
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- Gemfile	(date 1487374557000)
+++ Gemfile	(revision )
@@ -2,16 +2,18 @@
 
 # Bundle edge Rails instead: gem 'rails', github: 'rails/rails'
 gem 'rails', '4.1.11'
+
 # Use mysql as the database for Active Record in production
 gem 'mysql2', '~> 0.3.17'
 
 gem 'therubyracer',  platforms: :ruby
-# Sprockets (what Rails 3.1 uses for its asset pipeline) supports LESS
-gem 'sprockets', '2.11.0'
-gem 'less-rails', '~> 2.5.0'
-# https://github.com/seyhunak/twitter-bootstrap-rails
-gem 'twitter-bootstrap-rails', '3.2.0'
-# https://github.com/carrierwaveuploader/carrierwave
+
+# Boostrap with sass
+gem 'bootstrap-sass', '~> 3.2.0'
+gem 'autoprefixer-rails'
+
+
+# https://github.com/carrierwaveuploader/carrierwave Image uploader
 gem 'carrierwave'
 gem 'mini_magick'
 
Index: Gemfile.lock
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- Gemfile.lock	(date 1487374557000)
+++ Gemfile.lock	(revision )
@@ -37,6 +37,8 @@
       tzinfo (~> 1.1)
     addressable (2.4.0)
     arel (5.0.1.20140414130214)
+    autoprefixer-rails (6.7.3)
+      execjs
     bcrypt (3.1.11)
     bcrypt-ruby (3.1.5)
       bcrypt (>= 3.1.3)
@@ -49,6 +51,8 @@
     bootstrap-editable-rails (0.0.9)
       railties (>= 3.1)
       sass-rails
+    bootstrap-sass (3.2.0.2)
+      sass (~> 3.2)
     builder (3.2.2)
     byebug (9.0.5)
     cancan (1.6.10)
@@ -85,7 +89,6 @@
       coffee-script-source
       execjs
     coffee-script-source (1.10.0)
-    commonjs (0.2.7)
     database_cleaner (1.5.3)
     debug_inspector (0.0.2)
     devise (3.5.6)
@@ -127,11 +130,6 @@
     jquery-validation-rails (1.15.0.0)
     json (1.8.3)
     jwt (1.5.6)
-    less (2.5.1)
-      commonjs (~> 0.2.7)
-    less-rails (2.5.0)
-      actionpack (>= 3.1)
-      less (~> 2.5.0)
     libv8 (3.16.14.15)
     mail (2.6.4)
       mime-types (>= 1.16, < 4)
@@ -289,11 +287,6 @@
     turbolinks (5.0.0)
       turbolinks-source (~> 5)
     turbolinks-source (5.0.0)
-    twitter-bootstrap-rails (3.2.0)
-      actionpack (~> 4.1)
-      execjs (~> 2.2)
-      rails (~> 4.1)
-      railties (~> 4.1)
     tzinfo (1.2.2)
       thread_safe (~> 0.1)
     uglifier (3.0.1)
@@ -308,10 +301,12 @@
   ruby
 
 DEPENDENCIES
+  autoprefixer-rails
   bcrypt-ruby
   better_errors
   binding_of_caller
   bootstrap-editable-rails
+  bootstrap-sass (~> 3.2.0)
   cancan
   capistrano (~> 3.4.1)
   capistrano-bundler (~> 1.1.2)
@@ -330,7 +325,6 @@
   jbuilder (~> 2.0)
   jquery-rails
   jquery-validation-rails
-  less-rails (~> 2.5.0)
   mini_magick
   momentjs-rails
   mysql2 (~> 0.3.17)
@@ -351,12 +345,10 @@
   shoulda-matchers
   simplecov
   spring
-  sprockets (= 2.11.0)
   sqlite3
   therubyracer
   tilt
   turbolinks
-  twitter-bootstrap-rails (= 3.2.0)
   uglifier (>= 1.3.0)
   writeexcel
 
Index: app/assets/javascripts/application.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/assets/javascripts/application.js	(date 1487374557000)
+++ app/assets/javascripts/application.js	(revision )
@@ -12,7 +12,7 @@
 //
 //= require jquery
 //= require jquery_ujs
-//= require twitter/bootstrap
+//= require bootstrap-sprockets
 //= require undercore
 //= require gmaps/google
 //= require bootstrap-editable
\ No newline at end of file
Index: app/assets/stylesheets/home.css.less
===================================================================
--- app/assets/stylesheets/home.css.less	(date 1487374557000)
+++ app/assets/stylesheets/home.css.less	(date 1487374557000)
@@ -1,1 +0,0 @@
-
\ No newline at end of file
Index: app/assets/stylesheets/application.css.erb
===================================================================
--- app/assets/stylesheets/application.css.erb	(date 1487374557000)
+++ app/assets/stylesheets/application.css.sass	(revision )
@@ -20,3 +20,6 @@
 
 
  /* Your stylesheets goes here... */
+
+@import "bootstrap-sprockets"
+@import "bootstrap"
Index: app/assets/javascripts/home.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/assets/javascripts/home.js	(date 1487374557000)
+++ app/assets/javascripts/home.js	(revision )
@@ -47,4 +47,19 @@
 		$('#mai-escuela').fadeOut(); // Now the pop up is hiden.
 	});
 
+	$('.navbar-nav [data-toggle="tooltip"]').tooltip();
+	$('.navbar-twitch-toggle').on('click', function(event) {
+		event.preventDefault();
+		$('.navbar-twitch').toggleClass('open');
+	});
+
+	$('.nav-style-toggle').on('click', function(event) {
+		event.preventDefault();
+		var $current = $('.nav-style-toggle.disabled');
+		$(this).addClass('disabled');
+		$current.removeClass('disabled');
+		$('.navbar-twitch').removeClass('navbar-'+$current.data('type'));
+		$('.navbar-twitch').addClass('navbar-'+$(this).data('type'));
+	});
+
 })
Index: app/views/layouts/layout.html.erb
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/views/layouts/layout.html.erb	(date 1487374557000)
+++ app/views/layouts/layout.html.erb	(revision )
@@ -39,14 +39,61 @@
     <%= javascript_include_tag "application" %>
   </head>
   <body>
-	  <div class="container-fluid">
+		<div class="navbar navbar-default navbar-twitch" role="navigation">
+			<div class="container">
+				<div class="navbar-header">
+					<a class="navbar-brand" href="#">
+						<span class="small-nav"> <span class="logo">&nbsp;</span> </span>
+						<span class="full-nav"> < Bootsnipp > </span>
+					</a>
+				</div>
+				<div class="">
+					<ul class="nav navbar-nav">
+						<li class="active">
+							<a href="#">
+							<span class="small-nav" data-toggle="tooltip" data-placement="right" title="Home">
+								<span class="glyphicon glyphicon-home"></span>
+							</span>
+								<span class="full-nav"> Home </span>
+							</a>
+						</li>
+						<li>
+							<a href="#about-us">
+							<span class="small-nav" data-toggle="tooltip" data-placement="right" title="About Us">
+								<span class="fa fa-users"></span>
+							</span>
+								<span class="full-nav"> About Us </span>
+							</a>
+						</li>
+						<li>
+							<a href="#contact-us">
+							<span class="small-nav" data-toggle="tooltip" data-placement="right" title="Contact Us">
+								<span class="glyphicon glyphicon-comment"></span>
+							</span>
+								<span class="full-nav"> Contact Us </span>
+							</a>
+						</li>
+					</ul>
+				</div><!--/.nav-collapse -->
+			</div>
+		</div>
+		<button type="button" class="btn btn-default btn-xs navbar-twitch-toggle">
+			<span class="glyphicon glyphicon-chevron-right nav-open"></span>
+			<span class="glyphicon glyphicon-chevron-left nav-close"></span>
+		</button>
+		<div id="main-container" class="container-fluid">
+			<div class="row">
+				<div id="sidebar" class="col-lg-3">
+					sidebar
+				</div>
+				<div id="central-content" class="col-lg-9">
+					donde esas
+				</div>
+			</div>
 
-	    <%= bootstrap_flash %>
-	    <%= render 'layouts/header'%>
+			<%= render 'layouts/header'%>
 	    <%= yield %>
-
-
-	  </div> <!-- /container -->
+		</div> <!-- /container -->
 	  <%= render 'layouts/footer' %>
   </body>
 </html>
Index: app/assets/stylesheets/home.css.sass
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/assets/stylesheets/home.css.sass	(revision )
+++ app/assets/stylesheets/home.css.sass	(revision )
@@ -0,0 +1,144 @@
+@import url(//netdna.bootstrapcdn.com/font-awesome/4.0.3/css/font-awesome.css)
+
+.navbar-twitch
+  position: fixed
+  top: 0px
+  left: 0px
+  width: 50px
+  height: 100%
+  border-radius: 0px
+  border: 0px
+  z-index: 1030
+
+.navbar-twitch.open
+  width: 240px
+
+.navbar-twitch > .container
+  padding: 0px
+  width: 100%
+
+.navbar-twitch .navbar-header,
+.navbar-twitch .navbar-brand
+  float: none
+  display: block
+  width: 100%
+  margin: 0px
+
+.navbar-twitch .navbar-brand
+  height: 50px
+
+.navbar-twitch > .container .navbar-brand
+  padding: 0px
+  margin: 0px
+
+.navbar-twitch > .container .navbar-brand > .small-nav
+  display: block
+
+.navbar-twitch > .container .navbar-brand > .small-nav > span.logo,
+.navbar-twitch.open > .container .navbar-brand > .full-nav
+  display: block
+  font-weight: bold
+  padding: 15px 2px 15px 3px
+  font-size: 18pt
+
+.navbar-twitch.open > .container .navbar-brand
+  width: 100%
+  padding: 15px 15px 60px
+  text-align: center
+
+.navbar-twitch.navbar-inverse > .container .navbar-brand > .small-nav > span.logo
+  /*color: rgb(255, 255, 255)*/
+
+.navbar-twitch .navbar-nav > li
+  float: none
+
+.navbar-twitch > .container .full-nav
+  display: none
+
+.navbar-twitch.open > .container .full-nav
+  display: block
+
+.navbar-twitch.open > .container .small-nav
+  display: none
+
+.navbar-twitch .navbar-collapse
+  padding: 0px
+
+.navbar-twitch .navbar-nav
+  float: none
+  margin: 0px
+
+.navbar-twitch .navbar-nav > li > a
+  padding: 0px
+
+.navbar-twitch .navbar-nav > li > a > span
+  display: block
+  font-size: 16pt
+  padding: 13px 15px 13px 12px
+
+.navbar-twitch .tooltip,
+.navbar-twitch .tooltip .tooltip-inner
+  max-width: auto
+  white-space:nowrap
+
+.navbar-twitch-toggle
+  position: fixed
+  top: 5px
+  left: 55px
+
+.navbar-twitch ~ .navbar-twitch-toggle > .nav-open
+  display: inline-block
+
+.navbar-twitch ~ .navbar-twitch-toggle > .nav-close
+  display: none
+
+.navbar-twitch.open ~ .navbar-twitch-toggle
+  left: 245px
+
+.navbar-twitch.open ~ .navbar-twitch-toggle > .nav-open
+  display: none
+
+.navbar-twitch.open ~ .navbar-twitch-toggle > .nav-close
+  display: inline-block
+
+
+
+.btn-purple,
+.btn-purple:hover,
+.btn-purple:focus,
+.btn-purple:Active
+  color: rgb(255, 255, 255)
+  background-color: rgb(86, 61, 124)
+  border-color: rgb(111, 84, 153)
+
+.btn-purple:hover,
+.btn-purple:focus,
+.btn-purple:Active
+  background-color: rgb(111, 84, 153)
+  border-color: rgb(86, 61, 124)
+
+
+.navbar
+  background-image: none !important
+
+.navbar-purple
+  background-color: rgb(86, 61, 124)
+  border-color: rgb(111, 84, 153)
+
+.navbar-purple .navbar-brand,
+.navbar-purple .navbar-nav > li > a,
+.navbar-purple .navbar-brand:hover,
+.navbar-purple .navbar-nav > li > a:hover,
+.navbar-purple .navbar-brand:focus,
+.navbar-purple .navbar-nav > li > a:focus
+  color: rgb(205, 191, 227)
+
+.navbar-purple .navbar-brand:hover,
+.navbar-purple .navbar-nav > li > a:hover,
+.navbar-purple .navbar-brand:focus,
+.navbar-purple .navbar-nav > li > a:focus,
+.navbar-purple .navbar-nav > .active > a,
+.navbar-purple .navbar-nav > .active > a:hover,
+.navbar-purple .navbar-nav > .active > a:focus
+  background-color: rgb(111, 84, 153)
+  color: rgb(255, 255, 255)
Index: app/views/layouts/application.html.erb
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/views/layouts/application.html.erb	(date 1487374557000)
+++ app/views/layouts/application.html.erb	(revision )
@@ -5,11 +5,6 @@
   <%= javascript_include_tag "application", params[:controller] %>
 </head>
 <body>
-<%= link_to cart_path do%>
-  <i class="fi-shopping-cart"></i>
-  Carrito
-  (<span class="cart-count"><%=session[:product_ids].count%></span>)
-<%end%>
 
 <%= yield %>
 
